subprojects/core/src/main/java/org/gradle/process/internal/worker/DefaultWorkerProcess.java

```diff
@@ -227,6 +233,11 @@ public ExecResult waitForStop() {
         }
     }
 
+    @Override
+    public Optional<ExecResult> getExecResult() {
+        return Optional.ofNullable(execHandle.getExecResult());
```

Comment:
🤔 It might be better for `execHandle.getExecResult()` to just return `Optional<ExecResult>` instead of being nullable to avoid this mapping here.

Reply:
Unfortunately, ExecHandle has to be Java 6 compliant, so `Optional` isn't available in that interface.